homeassistant/components/garmin_connect/sensor.py

```diff
@@ -0,0 +1,186 @@
+    GarminConnectTooManyRequestsError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_ID
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import ATTRIBUTION, DOMAIN, GARMIN_ENTITY_LIST
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up Garmin Connect sensor based on a config entry."""
+    garmin_data = hass.data[DOMAIN][entry.entry_id]
+    unique_id = entry.data[CONF_ID]
+
+    try:
+        await garmin_data.async_update()
+    except (
+        GarminConnectConnectionError,
+        GarminConnectAuthenticationError,
+        GarminConnectTooManyRequestsError,
+    ) as err:
+        _LOGGER.error("Error occured during Garmin Connect Client update: %s", err)
+    except Exception:  # pylint: disable=broad-except
+        _LOGGER.error("Unknown error occured during Garmin Connect Client update.")
+
+    entities = []
+    for (
+        sensor_type,
+        (name, unit, icon, device_class, enabled_by_default),
+    ) in GARMIN_ENTITY_LIST.items():
+
+        _LOGGER.debug(
+            "Registering entity: %s, %s, %s, %s, %s, %s",
+            sensor_type,
+            name,
+            unit,
+            icon,
+            device_class,
+            enabled_by_default,
+        )
+        entities.append(
+            GarminConnectSensor(
+                garmin_data,
+                unique_id,
+                sensor_type,
+                name,
+                unit,
+                icon,
+                device_class,
+                enabled_by_default,
+            )
+        )
+
+    async_add_entities(entities, True)
+
+
+class GarminConnectSensor(Entity):
+    """Representation of a Garmin Connect Sensor."""
+
+    def __init__(
+        self,
+        data,
+        unique_id,
+        sensor_type,
+        name,
+        unit,
+        icon,
+        device_class,
+        enabled_default: bool = True,
+    ):
+        """Initialize."""
+        self._data = data
+        self._unique_id = unique_id
+        self._type = sensor_type
+        self._name = name
+        self._unit = unit
+        self._icon = icon
+        self._device_class = device_class
+        self._enabled_default = enabled_default
+        self._available = True
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return f"{DOMAIN}_{self._unique_id}_{self._type}"
```

Comment:
We don't need to include `DOMAIN` in the unique_id. The entity registry is aware of both the integration domain and the platform domain.